iothub/device/samples/getting started/FileUploadSample/Program.cs

```diff
@@ -34,13 +35,42 @@ public static async Task<int> Main(string[] args)
                     Environment.Exit(1);
                 });
 
-            using var deviceClient = DeviceClient.CreateFromConnectionString(
-                parameters.PrimaryConnectionString,
-                parameters.TransportType);
-            var sample = new FileUploadSample(deviceClient);
-            await sample.RunSampleAsync();
+            // If both IoT hub and device connection strings exist, default behavior uses existing device
+            if (parameters.DeviceConnectionString != null)
+            {
+                // run sample like normal
+                using var deviceClient = DeviceClient.CreateFromConnectionString(
+                    parameters.DeviceConnectionString,
+                    parameters.TransportType);
 
-            await deviceClient.CloseAsync();
+                var sample = new FileUploadSample(deviceClient);
+                await sample.RunSampleAsync();
+
+                await deviceClient.CloseAsync();
+            } 
+            else if (parameters.HubConnectionString != null)
+            {
+                // Deploy new device to IoT hub
+                using var registryManager = RegistryManager.CreateFromConnectionString(parameters.HubConnectionString);
+
+                string deviceId = $"FileUploadSample_{Guid.NewGuid()}";
```

Comment:
Let's get and set deviceId as a parameter

Reply:
Do we have other samples that allow users to pass in a desired device Id as a parameter? Also, would it make sense to allow a user to specify a device Id if it's just going to be deleted at the end of the sample?

Reply:
https://github.com/Azure/azure-iot-sdk-csharp/blob/previews/v2/provisioning/service/samples/Getting%20Started/IndividualEnrollmentX509Sample/Parameters.cs#L29

Reply:
Ohh we are removing the device after, makes sense

Reply:
most tests has a GUID as the deviceID.
what about a generic one like this?
nameof(<testname>)_{Guid.NewGuid()}